### ISSUE TITLE
Incorrectly inferred type when using fallback operator

### ISSUE THREAD

**Issue by zdila** [microsoft/TypeScript#35571]
**TypeScript Version:**  3.7.2

**Search Terms:**

fallback operator typescript infer wrong

**Code**

```ts
const index: number | null | undefined =
    Math.random() > 0.66 ? 1 : Math.random() > 0.33 ? null : undefined;

const arr = ['a', 'b'];

const value = (index ?? null) === null ? 'null/undefined' : arr[index];
```

**Expected behavior:**

No error should be produced.

**Actual behavior:**

Error for `index` in `arr[index]`:

```
Type 'null' cannot be used as an index type.ts(2538)
Type 'undefined' cannot be used as an index type.ts(2538)
```

**Comment by RyanCavanaugh** [comment]
We'd need negated types to handle this correctly; either way this is just an overcomplicated way to write `index == null ? 'null/undefined' : arr[index]` so there's not much upside to fixing it.
